Dockerfile

```diff
@@ -4,9 +4,30 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # BUILDPLATFORM is an automatic platform ARG enabled by Docker BuildKit.
-# Represents the plataform where the build is happening, do not mix with
+# Represents the platform where the build is happening, do not mix with
 # TARGETARCH
+
+# Stage 1: install dnsperf and its dependencies
+FROM docker.io/library/alpine:3.18.2@sha256:82d1e9d7ed48a7523bdebc18cf6290bdb97b82302a8a9c27d4fe885949ea94d1 as dnsperf-builder
+
+ARG DNSPERF_VERSION=2.13.0
+
+RUN apk update \
+    && apk add curl tar \
+    && apk add ldns-dev ck-dev build-base gcc libtool nghttp2-dev
+
+# Install dnsperf
+RUN curl -L https://www.dns-oarc.net/files/dnsperf/dnsperf-${DNSPERF_VERSION}.tar.gz --output dnsperf-${DNSPERF_VERSION}.tar.gz \
+    && tar zxvf dnsperf-${DNSPERF_VERSION}.tar.gz \
+    && cd dnsperf-${DNSPERF_VERSION}  \
+    && ./configure \
+    && make \
+    && make install
+
+# Stage 2: install needed binaries
 FROM docker.io/library/alpine:3.18.2@sha256:82d1e9d7ed48a7523bdebc18cf6290bdb97b82302a8a9c27d4fe885949ea94d1
+# Copy dnsperf binary from the previous stage
+COPY --from=dnsperf-builder /usr/local/bin/dnsperf /usr/bin/dnsperf
+RUN apk add --no-cache curl iputils bind-tools tcpdump ldns
```

Comment:
Checking the dependencies of the final build:
```console
$ docker run --entrypoint sh -ti --rm alpine-curl
/ # ldd /usr/bin/dnsperf 
	/lib/ld-musl-x86_64.so.1 (0x7f118e15d000)
	libssl.so.3 => /lib/libssl.so.3 (0x7f118e0b3000)
	libcrypto.so.3 => /lib/libcrypto.so.3 (0x7f118dca0000)
	libldns.so.3 => /usr/lib/libldns.so.3 (0x7f118dc47000)
	libnghttp2.so.14 => /usr/lib/libnghttp2.so.14 (0x7f118dc23000)
	libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7f118e15d000)
/ # apk info --who-owns /lib/libssl.so.3
/lib/libssl.so.3 is owned by libssl3-3.1.1-r1
/ # apk info --who-owns /lib/libcrypto.so.3
/lib/libcrypto.so.3 is owned by libcrypto3-3.1.1-r1
/ # apk info --who-owns /usr/lib/libldns.so.3
/usr/lib/libldns.so.3 is owned by ldns-1.8.3-r2
/ # apk info --who-owns /usr/lib/libnghttp2.so.14
/usr/lib/libnghttp2.so.14 is owned by nghttp2-libs-1.53.0-r0
# 
```

So it seems that the binary depends on `libssl3`, `libcrypto3`, `ldns` and `nghttp2-libs`. At the moment, they all seem to be pulled in by other dependencies (e.g. `bind-tools` pulls in `nghttp2-libs`), but if we ever remove some of the other dependencies here (e.g. `bind-tools`), then suddenly dnsperf will also break.

Can you add all those mentioned `dnsperf` dependencies explicitly and add a comment that they are needed for dnsperf? Because otherwise it will not be clear what dependencies we actually need for it, and where they are coming from. Thanks!